src/ksm/client/discovery.go

```diff
@@ -118,12 +117,20 @@ func (sd *discoverer) Discover(timeout time.Duration) (client.HTTPClient, error)
 	}
 
 	sd.logger.Debugf("KSM client created with endpoint=%v and nodeIP=%v", endpoint, nodeIP)
-	return newKSMClient(timeout, nodeIP, endpoint, sd.logger, sd.k8sClient), nil
+
+	KSMClient, err := newKSMClient(timeout, nodeIP, endpoint, sd.logger)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create KSM http client: %w", err)
+	}
+
+	return KSMClient, nil
 }
 
-func newKSMClient(timeout time.Duration, nodeIP string, endpoint url.URL, logger *logrus.Logger, k8s client.Kubernetes) *ksm {
-	bearer := k8s.Config().BearerToken
-	rt := newBearerRoundTripper(bearer)
+func newKSMClient(timeout time.Duration, nodeIP string, endpoint url.URL, logger *logrus.Logger) (*ksm, error) {
+	rt, err := transport.New(&transport.Config{TLS: transport.TLSConfig{Insecure: true}})
```

Comment:
I'm getting a bit lost with the imports here. To which object or package does `transport.` refer to here?

Reply:
`"k8s.io/client-go/transport"`

Reply:
Got it, from the docs I can also see that it should not include any extra headers in the request unless explicitly told to (which we aren't). LGTM.